src/main/java/com/example/moviemanager/controller/MovieRestController.java

```diff
@@ -0,0 +1,58 @@
+package com.example.moviemanager.controller;
+
+import com.example.moviemanager.entity.Movie;
+import com.example.moviemanager.service.MovieService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/movies")
+public class MovieRestController {
+
+    private final MovieService movieService;
+
+    @Autowired
+    public MovieRestController(MovieService movieService) {
+        this.movieService = movieService;
+    }
+
+    @GetMapping
+    public List<Movie> findAll() {
+        return movieService.findAll();
+    }
+
+    @GetMapping("/{movieId}")
+    public Movie findById(@PathVariable int movieId) {
+        return movieService.findById(movieId);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Movie addMovie(@Valid @RequestBody Movie movie) {
+        return movieService.save(movie);
+    }
+
+    @PutMapping
+    public Movie updateMovie(@Valid @RequestBody Movie movie) {
+        return movieService.save(movie);
```

Comment:
What if you try to update non existing movie? It would be saved as if it is new. I think you should specify which movie you are updating in URI. What is more, are all fields updatable? If no, then specific logic for updating entity should be introduced